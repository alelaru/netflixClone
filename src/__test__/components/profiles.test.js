import { getByText, render } from "@testing-library/react";
import Profiles from "../../components/profiles";


describe("<Profile/>", () => {

    it("Renders the <Profile/> with populated data", () => {

        const {container, getByTestId, getByText} = render(

        <Profiles>
            <Profiles.Title>Who's watching</Profiles.Title>
            <Profiles.List>
                <Profiles.User onClick={() => {}}>
                    <Profiles.Picture src={"images/karl.png"} data-testid="profile-picture"></Profiles.Picture>
                    <Profiles.Name>Karl</Profiles.Name>
                </Profiles.User>
            </Profiles.List>
        </Profiles>


        ) 
        expect(getByText("Who's watching")).toBeTruthy();
        expect(getByTestId("profile-picture")).toBeTruthy();
        expect(getByText("Karl")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    })
    it("Renders the <Profile/> with populated data bus misc profile image", () => {

        const {container, getByTestId, getByText} = render(

        <Profiles>
            <Profiles.Title>Who's watching</Profiles.Title>
            <Profiles.List>
                <Profiles.User onClick={() => {}}>
                    <Profiles.Picture src={"images/karl.png"} data-testid="profile-picture-misc"></Profiles.Picture>
                    <Profiles.Name>Karl</Profiles.Name>
                </Profiles.User>
            </Profiles.List>
        </Profiles>


        ) 
        expect(getByText("Who's watching")).toBeTruthy();
        expect(getByTestId("profile-picture-misc")).toBeTruthy();
        expect(getByText("Karl")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    })


})