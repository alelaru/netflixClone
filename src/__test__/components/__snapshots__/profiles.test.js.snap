// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> Renders the <Profile/> with populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 eJHeUh"
>
  <h1
    class="profiles__Title-bxtkkv-1 gEhjfN"
  >
    Who's watching
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 ibPFSe"
  >
    <li
      class="profiles__Item-bxtkkv-5 fVboEs"
    >
      <img
        class="profiles__Picture-bxtkkv-4 jOhiSx"
        data-testid="profile-picture"
        src="/images/users/images/karl.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 bLwqOe"
      >
        Karl
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profile/> Renders the <Profile/> with populated data bus misc profile image 1`] = `
<div
  class="profiles__Container-bxtkkv-0 eJHeUh"
>
  <h1
    class="profiles__Title-bxtkkv-1 gEhjfN"
  >
    Who's watching
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 ibPFSe"
  >
    <li
      class="profiles__Item-bxtkkv-5 fVboEs"
    >
      <img
        class="profiles__Picture-bxtkkv-4 jOhiSx"
        data-testid="profile-picture-misc"
        src="/images/users/images/karl.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 bLwqOe"
      >
        Karl
      </p>
    </li>
  </ul>
</div>
`;
